src/nodejs/neo-rpc/v2.9.0/client/module/getNodesBy.js

```diff
@@ -0,0 +1,344 @@
+                resolve(rankedList)
+              }
+            })
+            .catch(error => {
+              i++
+              print(__filename + ': getNodesBy.connections().error: ' + error)
+
+              if (i === nodes.length) {
+                rankedList = _.sortBy(rankedList, 'connections')
+                if (options.order === 'dsc') rankedList = rankedList.reverse()
+                resolve(rankedList)
+              }
+            })
+          }
+        })
+      }
+    })
+    .catch (error => {
+      print(__filename + ': getNodesBy.connections()/getNodesBy.ping().error: ' + error)
+    })
+  })
+}
+
+
+// Return an array of nodes sorted by version
+// ARGS:
+//  options = {
+//    net: 'TestNet',                                 // default network
+//    nodes: [{ 'url': 'https://host.domain:port' }], // defaults to empty
+//    order: 'dsc'                                    // asc = lowest value first, dsc = highest first
+//  }
+// TODO: add caching with configurable time limit for results
+// This will ALWAYS ping first with getNodesBy.ping()
+
+exports.version = (options) => {
+  let rankedList = [], i = 0, pingOptions = {}
+  Object.assign(options, pingOptions)
+  pingOptions.order = 'asc'
+
+  return new Promise((resolve, reject) => {
+    this.ping(pingOptions).then(nodes => {
+      if (_.isArray(nodes)) {
+        nodes.forEach((n) => {
+          if (n.url) {
+            const client = neon.default.create.rpcClient(n.url)
+
+            client.getVersion().then(response => {
+              i++
+              if (response) rankedList.push({ "url": n.url, "version": response })
+
+              if (i === nodes.length) {
+                rankedList = _.sortBy(rankedList, 'version')
+                if (options.order === 'dsc') rankedList = rankedList.reverse()
+                resolve(rankedList)
+              }
+            })
+            .catch(error => {
+              i++
+              print(__filename + ': getNodesBy.version().error: ' + error)
+
+              if (i === nodes.length) {
+                rankedList = _.sortBy(rankedList, 'version')
+                // if (options.order === 'dsc') rankedList = rankedList.reverse()
+                resolve(rankedList)
+              }
+            })
+          }
+        })
+      }
+    })
+    .catch (error => {
+      print(__filename + ': getNodesBy.version()/getNodesBy.ping().error: ' + error)
+    })
+  })
+}
+
+
+// Return an array of nodes sorted by getrawmempool
+// ARGS:
+//  options = {
+//    net: 'TestNet',                                 // default network
+//    nodes: [{ 'url': 'https://host.domain:port' }], // defaults to empty
+//    order: 'asc'                                    // asc = lowest value first, dsc = highest first
+//  }
+// TODO: add caching with configurable time limit for results
+// This will ALWAYS ping first with getNodesBy.ping()
+
+exports.rawmempool = (options) => {
```

Comment:
camelCase?

Reply:
This was intentional this time since the CLI calls the method by require(getnodesBy)[program.method]. I lowercased them to match the Neo API schema. I'm rethinking this. I may actually rewrite to camel case the user-supplied argument.

Reply:
Actually, I don't see a point. I'm going to leave it.